server/index.js

```diff
@@ -33,6 +36,37 @@ app.get('/api/entries/:userId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+/**
+ * route returns a single entry
+*/
+app.get('/api/entry/:entryId', (req, res, next) => {
+
+  const sql = `select "entryId", "amount", "description", TO_CHAR("date" :: DATE, 'mm/dd/yyyy') as "date", "name" as "category"
+               from "entry" join "category" using ("categoryId")
+               where "entryId" = $1`;
+
+  const entryId = parseInt(req.params.entryId, 10);
+
+  if (!entryId) {
+    throw new ClientError(400, 'EntryId is required');
+  }
+
+  if (entryId < 1 || !Number.isInteger(entryId)) {
+    throw new ClientError(400, 'EntryId must be a valid number');
+  }
+
+  const param = [entryId];
+
+  db.query(sql, param)
+    .then(singleEntry => {
+      if (!singleEntry.rows[0]) {
+        throw new ClientError(404, `Cannot find entry with entryID ${entryId}`);
+      }
+      res.status(200).json(singleEntry.rows);
```

Comment:
The client asked for one, so best give it to them and not give them an array containing one.